### ISSUE TITLE
Getting Started - working on this repo

### ISSUE THREAD

**Issue by richard-lane** [JGIBristol/Zebrafish_Osteoarthritis#6]
Start here if you are beginning to work on this project.

The other Issues are for specific things that I meant to fix/work on but didn't have time to get round to - this one is a general guide for getting started with the project. Talk also to the PI who will tell you what the overall plan is.

Overview
====
The shape and mechanical properties of bones is affected by lots of things (genes, sex, age, ...) - we want to understand these.

You can put a fish in a CT scanner and make a 3D image of it. If you were interested in e.g. the shape of the jaw, you could get a PhD to sit at a computer and segment out the bone from the surrounding tissue in e.g. ImageJ. This takes ages and is annoying, so it's not a feasible way to build up a dataset of jaws. You can instead train a machine learning model to do this.

Getting started
----
All the instructions should be in the README file, but if there's anything missing add it.

The "user interface" is the scripts in `scripts/` - these are documented pretty comprehensively in `docs/scripts/`.

The meat of the code is in `src/` - this is a library of helper functions/classes. You'll need to edit this/add new files etc. when you add new features (see the other issues).

There were some tests at one point but I might have forgotten to put them back in when I copied the code over from `zebrafish_jaw_segmentation`.

Set up your workspace
----
You'll need to decide where to run this code. I've been using `scampi`, which is a machine you should be able to get access to by asking your PI. Development locally is ok, but you won't be able to run some stuff and it might be irritating to access the data on the RDSF.

Before anythign else, you might want to fork this repository so you have your own copy. This isn't necessary, but might make you feel less scared about accidentally breaking something. (NB im not entirely sure how forks and `git-lfs`, which we use for storing the model weights, interact... maybe ask someone else in the team about this).

You will need to set up git lfs, then clone the repository from GitHub - see the README for instructions on how to do this. Set up SSH keys between wherever you will be working and GitHub, if you haven't already.

Your first step should be to clone the repository, install uv and run uv sync (see the README) to set up the environment and open a notebook. Once you have done that, you are ready to get started.

To do anything useful, you will also need to mount the RDSF - there are instructions in the README on how to do this.

Python Environment
----
If you've followed the previous steps, you should have a working python environment. Run the first script (from the command line) with the `--help` option just to check everything is working:

```
uv run scripts/0-create_dicoms.py -h
```

This should print some help output.

Once you have this set up, you should be able to run all the scripts. I suggest starting with the inference notebook - it is relatively complicated, so take some time to read the documentation in `docs/`. When you have done that, try performing inference on one of the scans. To do this:
1. Remember where you've mounted the RDSF
2. Find a CT scan to look at. One is on the RDSF at `DATABASE/uCT/Wahab_clean_dataset/TIFS/ak_1.tif`
3. Look at it with ImageJ or some other 3D image viewing software
4. Run something like `uv run scripts/3-run_inference.py example_locator example_segmenter.pkl ~/MY_RDSF_MOUNT/DATABASE/uCT/Wahab_clean_dataset/TIFS/ak_1.tif --output-dir 3d_tif_example`, replacing `MY_RDSF_MOUNT` with the right RDSF mount point to run the segmentation
5. Check you have some output in `3d_tif_example/`
6. Open it in ImageJ or some other program for opening 3D images - check that the segmentation mask looks sensible and that you understand what's going on

Once you've done this, you're in position to run segmentation on lots of jaws. See if you can do this.

You might then want to check everything works by running all the scripts (there aren't too many).

If you've done that, everything worked and you understand what's happening, you might want to play around on some final stuff:
- check if you can train a jaw locating model. Look at the plots it spits out, see if you understand them.
- check if you can change the segmentation model hyperparams (learning rate, etc.) and see how that affects the model

If you've done this, you're in position to start working on the other Issues in this repo. good luck
